scripts/corelli/calibration/database.py

```diff
@@ -7,118 +7,198 @@
+def filename_bank_table(bank_id: int, database_path: str, date: str, table_type: str = 'calibration') -> str:
     """
     Function that returns the absolute path for a bank file using corelli format:
     database_path/bank0ID/type_corelli_bank0ID_YYYYMMDD.nxs
-    :param bankID bank number that is calibrated
+    :param bank_id bank number that is calibrated
     :param date format YYYYMMDD
     :param database_path location of the corelli database (absolute or relative)
            Example: database/corelli/ for
                     database/corelli/bank001/
                     database/corelli/bank002/
-    :param type 'calibration', 'mask' or 'fit'
+    :param table_type: 'calibration', 'mask' or 'fit'
     """
-
-    verify_date_YYYYMMDD('filename_bank_table', date)
-    subdirectory:str = database_path + '/' + 'bank' + str(bankID).zfill(3)
+    TableType.assert_valid_type(table_type)
+    verify_date_format('filename_bank_table', date)
+    subdirectory: str = database_path + '/' + 'bank' + str(bank_id).zfill(3)
     pathlib.Path(subdirectory).mkdir(parents=True, exist_ok=True)
-    message = f'{"Cannot process Corelli filename_bank_table, table_type must be calibration, mask or fit"}'
-    assert table_type == 'calibration' or table_type == 'mask' or table_type == 'fit', message
 
-    filename = subdirectory + '/' + table_type + '_corelli_bank' + str(bankID).zfill(3) + '_' + date + '.nxs.h5'
+    filename = subdirectory + '/' + table_type + '_corelli_bank' + str(bank_id).zfill(3) + '_' + date + '.nxs.h5'
     # make it portable
     filename = str(pathlib.Path(filename).resolve())
     return filename
 
 
-def save_manifest_file( bankIDs:list, database_path:str, date:str)->None:
+def save_manifest_file(database_path: str,
+                       bank_ids: List[Union[int, str]],
+                       day_stamps: List[Union[int, str]],
+                       manifest_day_stamp: Optional[Union[int, str]] = None) -> str:
     """
-    Function that saves or updates an existing manifest_corelli_date.csv file.
-    There is one file stored for each calibration day. If file exist it will append two columns:
-    bankID, timestamp (ISO format)
-    :param bankIDs input of bankIDs that have been calibrated
-    :param database_path location of the corelli database (absolute or relative)
-           Example: database/corelli/ for manifest file:
-                    database/corelli/manifest_corelli_YYYYMMDD.csv
-    :param date format YYYYMMDD
+    Function that saves a manifest_corelli_date.csv file.
+    There is one file stored for each time a new new correlli calibration is invoked. The file
+    consists of two columns: bank_id, timestamp (ISO format)
+
+    :param database_path: location of the corelli database (absolute or relative)
+    :param bank_ids: input of bank numbers that have been calibrated
+    :param day_stamps: day stamp for each of the calibrated banks
```

Comment:
Similarly, explain the Union components, are they the same content just different types (int,string)?

Reply:
Yes, it's the same content with different types. I now explain this in the dostring